src/dparse/stack_buffer.d

```diff
@@ -20,14 +22,21 @@ struct StackBuffer
         static if (is(T == class))
             static assert(T.sizeof == size_t.sizeof);
 
+        static if (hasIndirections!T)
+            while (_length % size_t.sizeof != 0)
+                put(ubyte(1));
+
         if (arr.ptr != stackSpace.ptr)
         {
             if (_length + T.sizeof > arr.length)
             {
                 size_t newLength = arr.length << 1;
                 while (_length + T.sizeof > newLength)
                     newLength <<= 1;
+                auto oldPtr = arr.ptr;
                 Mallocator.instance.reallocate(arr, newLength);
+                GC.removeRange(oldPtr);
```

Comment:
I think there is a race condition here, which is unavoidable with the `reallocate` API.

We need to call `removeRange` after the data is copied, but before the old buffer is deallocated. `reallocate` does both.

A bit surprising that `stdx.allocator` doesn't have a layer which calls `addRange`/`removeRange`, though maybe that kind of defeats the point of using custom allocators.

Also note that we can't make this call conditional on `hasIndirections!T` because `T` is just for the value being allocated right now, we don't know what the rest of the buffer contains, nor can we know if the buffer will be completely used for types without indirections.